tests/test_forum.py

```diff
@@ -262,30 +263,40 @@ async def test_edit_general_forum_topic(self, bot, forum_group_id):
         assert result is True, "Failed to edit general forum topic"
         # no way of checking the edited name, just the boolean result
 
-    async def test_close_and_reopen_general_forum_topic(self, bot, forum_group_id):
+    async def test_close_reopen_hide_unhide_general_forum_topic(self, bot, forum_group_id):
+        """Since reopening also unhides and hiding also closes, testing (un)hiding and
+        closing/reopening in different tests would mean that the tests have to be executed in
+        a specific order. For stability, we instead test all of them in one test."""
+
+        # We first ensure that the topic is open and visible
+        # Otherwise the tests below will fail
+        try:
+            await bot.reopen_general_forum_topic(chat_id=forum_group_id)
+        except BadRequest as exc:
+            if "Topic_not_modified" not in exc.message:
+                raise exc
+
+        # first just close, bot don't hide
         result = await bot.close_general_forum_topic(
             chat_id=forum_group_id,
         )
         assert result is True, "Failed to close general forum topic"
 
-        result = await bot.reopen_general_forum_topic(
-            chat_id=forum_group_id,
-        )
-        assert result is True, "Failed to reopen general forum topic"
-
-    async def test_hide_and_unhide_general_forum_topic(self, bot, forum_group_id):
-
+        # then hide
         result = await bot.hide_general_forum_topic(
             chat_id=forum_group_id,
         )
         assert result is True, "Failed to hide general forum topic"
 
+        # then unhide, bot don't reopen
         result = await bot.unhide_general_forum_topic(
             chat_id=forum_group_id,
         )
         assert result is True, "Failed to unhide general forum topic"
 
-        # hiding the general topic also closes it, so we reopen it
+        # finally, reopen
+        # as this also unhides, this should ensure that the topic is open and visible
+        # for the next test run
```

Comment:
If we create a fixture with the initial reopening (above, with catching of the exception) of the general forum topic, then `yield`, and then this final re-opening, then we can ensure that the general topic remains open even if for some reason this test fails halfway through.

Maybe it will too much code for a simple test like this, though. Plus this fixture is unlikely to be reused elsewhere.

Reply:
If we were to use the general topic in more tests, I would definitely agree :) While we only have this one test, I think it's more readable this way.